### PR TITLE
fix(sdcm/coredump): Move out coredump code and make it's shutdown better

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/t6CIu9jH/2136-job-set-as-passed-while-coredump-were-found-and-nemesis-failed

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by bentsi** [comment]
@dkropachev please add a description and a link to Trello

**Comment by dkropachev** [comment]
> Can you point me to the mechanism that waits for the coredump upload to complete?

It is in the def run(self), but there is one problem currently we wait only 2 minutes for it to complete, as you know cores could be up to 50gb, not possible to get them uploaded in this time period.  Should we increase it ?

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1135,27 +939,53 @@ def start_task_threads(self):
...
+            await_bucket.append(self._coredump_thread)
+        if self._decoding_backtraces_thread:
+            await_bucket.append(self._decoding_backtraces_thread)
+        end_time = time.perf_counter() + timeout
+        while await_bucket and end_time > time.perf_counter():
+            for thread in await_bucket.copy():
+                if not thread.is_alive():
+                    await_bucket.remove(thread)
+            time.sleep(1)
+
```
Comment (bentsi) [inline sdcm/cluster.py:983]: remove `_coredump_thread` from `await_bucket` and wait for upload to finsih

Comment (dkropachev) [reply sdcm/cluster.py:989]: done

**Comment by dkropachev** [comment]
Tested here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/393

**Review by bentsi — APPROVED** [review]

**Pull request merged**